### ISSUE TITLE
Train the anomaly detection model.

### ISSUE THREAD

**Issue by sastead** [zzjay-005/S25_CSS_04_ZS#19]
Apply supervised and unsupervised learning techniques.
Monitor loss function and optimize model hyperparameters.
